src/lobster/tokenization/_ume_tokenizers.py

```diff
@@ -461,71 +463,266 @@ def __init__(self):
         )
 
 
-class UmeTokenizerTransform(TokenizerTransform):
+def _get_modality_tokenizer(modality: ModalityType | str) -> PreTrainedTokenizerFast:
+    modality = Modality(modality) if isinstance(modality, str) else modality
+
+    match modality:
+        case Modality.AMINO_ACID:
+            tokenizer = UmeAminoAcidTokenizerFast()
+        case Modality.SMILES:
+            tokenizer = UmeSmilesTokenizerFast()
+        case Modality.NUCLEOTIDE:
+            tokenizer = UmeNucleotideTokenizerFast()
+        case Modality.COORDINATES_3D:
+            tokenizer = UmeLatentGenerator3DCoordTokenizerFast()
+
+    return tokenizer
+
+
+class UmeTokenizerTransform(Module):
+    """
+    Ume tokenizer transform for single or dual modality inputs.
+
+    For single modality:
+        Tokenizes inputs using the specified modality tokenizer
+        with vocabulary that's aware of reserved tokens.
+
+    For dual modality:
+        Combines inputs as: [CLS] [input1] [SEP] <task_token> [input2] [SEP]
+        where task_token is either <interact> or <convert>.
+
+    Examples:
+    ```
+    tokenizer = UmeTokenizerTransform(
```

Comment:
nit: for these kinds of inline code examples you might as well use follow the [doctest](https://docs.python.org/3/library/doctest.html) conventions. I think we already do this in the main `Ume` class. Probably worth adding these to ci at some point